credentials/tls/certprovider/pemfile/builder.go

```diff
@@ -62,34 +60,12 @@ func pluginConfigFromJSON(jd json.RawMessage) (Options, error) {
 	// The only difference between this anonymous struct and the Options struct
 	// is that the refresh_interval is represented here as a duration proto,
 	// while in the latter a time.Duration is used.
-	cfg := &struct {
```

Comment:
@easwars IMO, the existing way of using an anonymous struct seems better. It keeps the parsing logic contained in this single function. With the change in the PR we're adding our own Duration type and a method in another file, exposing the deserialization behaviour to all users of the `Options` struct. I wanted to get your thoughts on this.

Reply:
One way to reduce duplication is to embed the `Options` struct here:
```go
cfg := &struct {
    Options   `json:",inline"`
    RefreshInterval string `json:"refresh_interval,omitempty"`
}{}
```

The `RefreshInterval` of the anonymous struct would get the deserialized field from the json. We can then parse it and set the field on `cfg.Duration` before returning `cfg.Duration`.